iterative_assembly.wdl

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2018 Sequencing Analysis Support Core - Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import "tasks/bwa.wdl" as bwa
+import "tasks/samtools.wdl" as samtools
+import "tasks/spades.wdl" as spades
+import "tasks/biopet.wdl" as biopet
+import "tasks/seqtk.wdl" as seqtk
+import "tasks/picard.wdl" as picard
+
+# This workflow takes an existing assembly and reads that have already passed QC.
+# It maps the reads back to the assembly and uses the mapped reads to create a new assembly.
+# This workflow can be run on its own spades.contigs or spades.scaffolds output
+workflow iterativeAssembly {
```

Comment:
I think the name can be confusing, this it not really a iterative pipeline, just a single assembly step with extracted reads from a bam file.

If this is really iterative this process should repeat itself till no changes are detected or if a maximum number of iterations has been reached.

A better name could be ReAssembly, this could be a subworkflow in a iterative assembly pipeline ofcourse.

Reply:
Agreed